cmake/multi_image.cmake

```diff
@@ -125,6 +125,22 @@ function(add_child_image_from_source)
   set(${ACI_NAME}_BOARD ${ACI_BOARD})
 
   message("\n=== child image ${ACI_NAME} - ${ACI_DOMAIN}${inherited} begin ===")
+
+  # Look for a folder named 'child_image_overlay' and apply all relevant
+  # overlay files. An overlay file will be applied if it is located in the
+  # root of the child_image_overlay folder and matches ${ACI_NAME}-overlay.conf
+  # or if it is stored in
+  # 'child_image_overlay/parent_board_name/${ACI_NAME}-overlay.conf'
+  set(child_conf ${CMAKE_SOURCE_DIR}/child_image_overlay)
+  foreach (overlay_file
+      ${child_conf}/${ACI_NAME}-overlay.conf
+      ${child_conf}/${BOARD}/${ACI_NAME}-overlay.conf)
+    message("Checking ${overlay_file}")
+    if (EXISTS ${overlay_file})
+      add_overlay_config(${ACI_NAME} ${overlay_file})
```

Comment:
I believe it has been a mistake to use `OVERLAY_CONFIG` inside CMake, that should be reserved for the end-user.
We should instead construct a list of config file fragments and pass that to the child image, so that `-D<aci>_OVERLAY_CONFIG` is for user, and so he can overrule everything if specifying `-D<aci>_CONF_FILE`, just as it is possible for single image configurations.
Will also make everything more consistent == easier to explain.